freva-client/src/freva_client/query.py

```diff
@@ -201,13 +254,21 @@ def _add_search_keyword_args_from_facet(
 
         if facets and num_facets == 0:
             # TODO: This isn't pretty, but if a user requested a search
-            # string doesn't exist than we have to somehow make the search
+            # string that doesn't exist than we have to somehow make the search
             # return nothing.
             search_kw = {primary_key: ["NotAvailable"]}
         self._params.update(search_kw)
 
     def __iter__(self) -> Iterator[str]:
-        result = self._get(self._cfg.search_url)
+        query_url = self._cfg.search_url
+        headers = {}
+        if self._stream_zarr:
+            query_url = self._cfg.zarr_loader_url
+            auth = Auth()
+            auth.check_authentication(auth_url=self._cfg.auth_url)
+            token = cast(Token, auth.auth_token)
+            headers = {"Authorization": f"Bearer {token['access_token']}"}
+        result = self._get(query_url, headers=headers, stream=True)
```

Comment:
I am not entirely sure what you mean, to be honest. Could you clarify? Do you not like the code block you've mentioned? Thanks for clarifying.

Reply:
no, actually it's not an issue, just a minor comment to make the whole script in a same shape and then fast in reading ... 

If you look at these lines, in `_create_intake_catalogue_file` function you went for the same approach:
https://github.com/FREVA-CLINT/freva-nextgen/blob/stream-data/freva-client/src/freva_client/query.py#L350-L359

deserved to be ignored ...

Reply:
You mean this bit is missing?

```python
 if result is None:
            raise ValueError("No results found")
```

Searchs where nothing was found should still be valid.  Because this generates a list of found entries. If nothing was found then the list will we empty, that's fair enough. It should be up to the user using the client to handle the situation if nothing was found. 

On the other hand for the intake catalogue: I don't think we can create an empty intake catalogue.:

```python
Path("cat.json").write_text("{}", encoding="utf-8")
cat = intake.open_esm_datastore("cat.json")
ValidationError: 3 validation errors for ESMCatalogModel
esmcat_version
  Field required [type=missing, input_value={'last_updated': None}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
attributes
  Field required [type=missing, input_value={'last_updated': None}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
assets
  Field required [type=missing, input_value={'last_updated': None}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing

```

If you think it's worth creating an empty catalogue I can see how this is done.